src/main/java/com/google/api/generator/engine/ast/NewObjectValue.java

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import java.util.Collections;
+import java.util.List;
+
+@AutoValue
+public abstract class NewObjectValue implements ObjectValue {
+  public abstract TypeNode type();
+
+  public abstract List<Expr> arguments();
+
+  public static Builder builder() {
+    return new AutoValue_NewObjectValue.Builder().setArguments(Collections.emptyList());
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setType(TypeNode type);
+
+    public abstract Builder setArguments(List<Expr> arguments);
+    // abstract NewObjectValue autobuild();
+    // public static Builder builder(){
+
+    //     // there should be reference in typeNode
+    // }
+    public abstract NewObjectValue build();
+  }
+
+  @Override
+  public String value() {
+    StringBuilder value = new StringBuilder();
+    value.append("new ").append(type().reference().name());
```

Comment:
what if the reference is null.

Reply:
This will be checked in the build() in NewObjectValue, thanks!

Reply:
done.